### PR TITLE
feat: aes-2486: flyout panel heading

### PR DESCRIPTION
**Pull request by kevin-ho87**
**Ticket**
https://aesoponline.atlassian.net/browse/AES-2486

**Summary**
Update Hooks and FlyinPanel to control FlyinPanel heading from PDP

**Development tasks summary**
- `FlyinPanel` move heading into content div
- Update `FlyinPanel` test to check heading
- PDP add `FlyinPanel` heading text
- `flyinPanelHeading` copy from PDP passed into `FlyinPanel`

### CODE REVIEW

File: src/components/ProductDetailHeader/components/ProductDetailBody/ProductDetailBody.js [hunk 1]
```diff
@@ -232,6 +233,7 @@ const ProductDetailBody = ({ className, copy, theme }) => {
 
       {flyinPanel && (
         <FlyinPanel
+          heading={hasFlyinPanelHeading && copy.flyinPanelHeading}
```
Comment (zack76) [inline src/components/ProductDetailHeader/components/ProductDetailBody/ProductDetailBody.js:236]: is `hasFlyinPanelHeading` necessary in this case?

Comment (kevin-ho87) [reply src/components/ProductDetailHeader/components/ProductDetailBody/ProductDetailBody.js:236]: Yes, they want to control the display of the heading from Contentful.

Comment (kevin-ho87) [reply src/components/ProductDetailHeader/components/ProductDetailBody/ProductDetailBody.js:236]: image

**Review by zack76 — APPROVED** [review]

File: src/components/FlyinPanel/FlyinPanel.spec.js [hunk 2]
```diff
@@ -14,13 +14,15 @@ describe('<FlyinPanel />', () => {
     render(
       <FlyinPanel
         copy={{ close: 'close flyin panel button' }}
+        heading="Ingredients"
```
Comment (MkMan) [inline src/components/FlyinPanel/FlyinPanel.spec.js:17]: Can you please make this a new test, say "should show a heading if provided"? If you move it out, you can also add an assertion here that there is no Heading in the DOM if the `heading` prop is undefined. That way we cover more of our code 🙂

File: src/components/FlyinPanel/FlyinPanel.spec.js [hunk 3]
```diff
@@ -14,13 +14,15 @@ describe('<FlyinPanel />', () => {
...
         isVisible={true}
         onClose={handleOnClose}
       >
         flyin panel
       </FlyinPanel>,
     );
 
+    expect(screen.getByText(/ingredients/i)).toBeTruthy();
```
Comment (MkMan) [inline src/components/FlyinPanel/FlyinPanel.spec.js:25]: I think using `toBeInTheDocument()` over `toBeTruthy()` would be better here. It makes it easier to understand what it being asserted, an element being present in the DOM.

File: src/components/FlyinPanel/FlyinPanel.stories.mdx [hunk 4]
```diff
@@ -24,11 +24,8 @@ import { FlyinPanel } from './FlyinPanel';
               onClose={() => {
                 setState(() => ({ ...state, isVisible: false }));
               }}
-              title="Ingredients"
+              heading="Ingredients"
```
Comment (MkMan) [inline src/components/FlyinPanel/FlyinPanel.stories.mdx:27]: What was `title`? 🤔

File: src/components/ProductDetailHeader/components/ProductDetailBody/ProductDetailBody.js [hunk 5]
```diff
@@ -42,6 +42,7 @@ const ProductDetailBody = ({ className, copy, theme }) => {
   const {
     definitionList,
     flyinPanel,
+    hasFlyinPanelHeading,
```
Comment (MkMan) [inline src/components/ProductDetailHeader/components/ProductDetailBody/ProductDetailBody.js:45]: Is this prop necessary? Could it be inferred from `copy.flyinPanelHeading`?

File: src/components/FlyinPanel/FlyinPanel.stories.mdx (hunk 4 above)
Comment (kevin-ho87) [reply src/components/FlyinPanel/FlyinPanel.stories.mdx:27]: Im not sure, I couldnt find what `title` was being used for.

Comment (MkMan) [reply src/components/FlyinPanel/FlyinPanel.stories.mdx:27]: Fair enough. Maybe a prop that has been removed but missed inn the Story 🙂

File: src/components/ProductDetailHeader/components/ProductDetailBody/ProductDetailBody.js (hunk 5 above)
Comment (kevin-ho87) [reply src/components/ProductDetailHeader/components/ProductDetailBody/ProductDetailBody.js:45]: image
They wanted to be able to control toggling it in the Contentful model.
`Is Ingredients` will be used to toggle the Heading display.

Comment (kevin-ho87) [reply src/components/ProductDetailHeader/components/ProductDetailBody/ProductDetailBody.js:45]: Actually, Ill see If I can control it in web-ui, to pass in null if `is Ingredients` is false

File: src/components/FlyinPanel/FlyinPanel.spec.js (hunk 2 above)
Comment (kevin-ho87) [reply src/components/FlyinPanel/FlyinPanel.spec.js]: I have updated the test

File: src/components/FlyinPanel/FlyinPanel.spec.js (hunk 3 above)
Comment (kevin-ho87) [reply src/components/FlyinPanel/FlyinPanel.spec.js]: I have moved this out into its on `it` block and used `toBeInTheDocument()`

File: src/components/ProductDetailHeader/components/ProductDetailBody/ProductDetailBody.js (hunk 5 above)
Comment (kevin-ho87) [reply src/components/ProductDetailHeader/components/ProductDetailBody/ProductDetailBody.js]: I have removed the boolean check and added the logic on web-ui side

**Review by MkMan — APPROVED** [review]

**Pull request merged**